proj/hps_accel/gateware/stream/gearbox.py

```diff
@@ -0,0 +1,74 @@
+#!/bin/env python
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from nmigen import signed, unsigned, Cat, Signal, Module
+from nmigen.hdl.dsl import FSM
+from nmigen_cfu import SimpleElaboratable
+
+from .stream import Endpoint
+
+
+class ByteToWord(SimpleElaboratable):
+    """Converts a stream of bytes to a stream of words.
+
+    Attributes:
+    ----------
+
+    input: Endpoint(signed(8)), in
+      Incoming stream of bytes
+    output: Endpoint(unsigned(32)), out
+      Outgoing stream of words
+    """
+
+    def __init__(self):
+        self.input = Endpoint(signed(8))
+        self.output = Endpoint(unsigned(32))
+
+    def elab(self, m: Module):
+
+        # Registers accumulate incoming values
+        registers = [Signal(8, name="register{i}") for i in range(3)]
+        waiting_to_send = Signal()
```

Comment:
Neat. I probably needed to use some pattern like this in the store rewrite that I tried. I have made a note to think about this when I look at the stores again.